src/main/java/uk/ac/ebi/ena/webin/cli/manifest/processor/metadata/SampleProcessor.java

```diff
@@ -80,7 +81,7 @@ private void handleHttpError(HttpStatusCodeException ex, String sampleId) {
     switch (ex.getStatusCode()) {
       case UNAUTHORIZED:
       case FORBIDDEN:
-        throw WebinCliException.userError(WebinCliMessage.CLI_AUTHENTICATION_ERROR.text());
+        throw WebinCliException.userError(WebinCliMessage.SERVICE_AUTHENTICATION_ERROR.format(SampleService.SERVICE_NAME));
```

Comment:
Thought about this long and hard. We have (at least) two ways to handle HTTP errors. NotFoundErrorHandler and handleHttpError in SampleProcessor and SampleXmlProcessor that is a precursor for NotFoundErrorHandler. The most concise (and readable) way to handle HTTP errors is to use setErrorHander. See the example from StudyService:
``` 
        restTemplate.setErrorHandler(new NotFoundErrorHandler(
                WebinCliMessage.STUDY_SERVICE_VALIDATION_ERROR.format(studyId),
                WebinCliMessage.STUDY_SERVICE_SYSTEM_ERROR.format(studyId)));
```
I think we should improve NotFoundErrorHandler and use it everywhere including SampleProcessor and SampleXmlProcessor. Example for StudyService:

``` 
        restTemplate.setErrorHandler(new NotFoundErrorHandler(
                StudyService.SERVICE_NAME,
                WebinCliMessage.STUDY_SERVICE_VALIDATION_ERROR.format(studyId),
                WebinCliMessage.STUDY_SERVICE_SYSTEM_ERROR.format(studyId)));
```

```
/*
 * Copyright 2018-2021 EMBL - European Bioinformatics Institute
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
 * file except in compliance with the License. You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software distributed under the
 * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 */
package uk.ac.ebi.ena.webin.cli.service.handler;

import static org.springframework.http.HttpStatus.Series.CLIENT_ERROR;
import static org.springframework.http.HttpStatus.Series.SERVER_ERROR;

import java.io.IOException;

import org.springframework.http.client.ClientHttpResponse;
import org.springframework.web.client.ResponseErrorHandler;

import uk.ac.ebi.ena.webin.cli.WebinCliException;
import uk.ac.ebi.ena.webin.cli.WebinCliMessage;

public class NotFoundErrorHandler implements ResponseErrorHandler {

    private final String serviceName;
    private final String notFoundError;
    private final String systemError;

    public NotFoundErrorHandler(String serviceName, String notFoundError, String systemError) {
        this.serviceName = serviceName;
        this.notFoundError = notFoundError;
        this.systemError = systemError;
    }

    @Override
    public boolean hasError(ClientHttpResponse httpResponse) throws IOException {
        return (httpResponse.getStatusCode().series() == CLIENT_ERROR ||
                httpResponse.getStatusCode().series() == SERVER_ERROR);
    }

    @Override
    public void handleError(ClientHttpResponse httpResponse) throws IOException {
        switch (httpResponse.getStatusCode()) {
            case UNAUTHORIZED:
            case FORBIDDEN:
                throw WebinCliException.userError(WebinCliMessage.SERVICE_AUTHENTICATION_ERROR.format(serviceName));
            case NOT_FOUND:
                throw WebinCliException.validationError(notFoundError);
            default:
                throw WebinCliException.systemError(systemError);
        }
    }
}
```